src/main/java/dev/slimevr/vr/processor/skeleton/LegTweaks.java

```diff
@@ -0,0 +1,920 @@
+				.getLeftFootPositionCorrected(null).y;
+			if (
+				Math.abs(leftFootPosition.y - lastPositionY) < FOOT_Y_DIFF_CUTOFF
+					&& leftFootPosition.y > lastPositionY
+					&& bufferHead.getLeftFootAccelerationY() < FOOT_Y_MAX_ACCELERATION
+			) {
+				leftFootPosition.y = lastPositionY;
+				correctLeft = false;
+			}
+		}
+
+		if (bufferHead.getRightLegState() == LegTweakBuffer.LOCKED) {
+			float lastPositionY = bufferHead
+				.getParent()
+				.getRightFootPositionCorrected(null).y;
+
+			if (
+				Math.abs(rightFootPosition.y - lastPositionY) < FOOT_Y_DIFF_CUTOFF
+					&& rightFootPosition.y > lastPositionY
+					&& bufferHead.getRightFootAccelerationY() < FOOT_Y_MAX_ACCELERATION
+			) {
+				rightFootPosition.y = lastPositionY;
+				correctRight = false;
+			}
+		}
+
+		// using the velocity correct the position
+		if (correctLeft) {
+			correctLeftFootTrackerY();
+		}
+
+		if (correctRight) {
+			correctRightFootTrackerY();
+		}
+	}
+
+	private void correctLeftFootTrackerY() {
+		Vector3f temp;
+		Vector3f leftFootDif = leftFootPosition
+			.subtract(bufferHead.getParent().getLeftFootPositionCorrected(null));
+		if (Math.abs(leftFootDif.y) > NEARLY_ZERO) {
+			temp = bufferHead
+				.getParent()
+				.getLeftFootPositionCorrected(null)
+				.subtract(
+					bufferHead
+						.getParent()
+						.getLeftFootPosition(null)
+						.subtract(bufferHead.getLeftFootPosition(null))
+				);
+
+			float leftFloor = (floorLevel + (MAX_DYNAMIC_DISPLACEMENT * getLeftFootOffset()))
+				- currentDisengagementOffset;
+
+			if (
+				temp.y
+					< leftFloor
+			) {
+				leftFootPosition.y = leftFloor;
+			} else {
+				leftFootPosition.y = temp.y;
+			}
+
+			Vector3f velocity = bufferHead.getLeftFootVelocity(null);
+
+			if (velocity.y * leftFootDif.y > 0) {
+				leftFootPosition.y += velocity.y * FOOT_Y_CORRECTION_WEIGHT;
+			} else {
+				leftFootPosition.y -= velocity.y * FOOT_Y_CORRECTION_WEIGHT;
+			}
+
+			// check for overshoot and correct if necessary
+			if (
+				checkOverShoot(
+					this.bufferHead.getLeftFootPosition(null).y,
+					this.bufferHead.getParent().getLeftFootPositionCorrected(null).y,
+					leftFootPosition.y
+				)
+			) {
+				leftFootPosition.y = bufferHead.getLeftFootPosition(null).y;
+			}
+		}
+	}
+
+	private void correctRightFootTrackerY() {
+		Vector3f temp;
+		Vector3f rightFootDif = rightFootPosition
+			.subtract(bufferHead.getParent().getRightFootPositionCorrected(null));
+
+		if (Math.abs(rightFootDif.y) > NEARLY_ZERO) {
+			temp = bufferHead
+				.getParent()
+				.getRightFootPositionCorrected(null)
+				.subtract(
+					bufferHead
+						.getParent()
+						.getRightFootPosition(null)
+						.subtract(bufferHead.getRightFootPosition(null))
+				);
+
+			float rightFloor = (floorLevel + (MAX_DYNAMIC_DISPLACEMENT * getRightFootOffset()))
+				- currentDisengagementOffset;
+
+			if (
+				temp.y
+					< rightFloor
+			) {
+				rightFootPosition.y = rightFloor;
+			} else {
+				rightFootPosition.y = temp.y;
+			}
```

Comment:
Why not something like this?
```suggestion
			rightFootPosition.y = temp.y < rightFloor : temp.y;
```
(if you decide on going for this, please do it in the other places as well)

Reply:
I am going with that for any if else blocks that are simple like that. As far as I'm aware it's just the two in the right and left versions of this function. I can change more than that but they are more complicated requiring this:

'if (velocity.y * leftFootDif.y > 0) {
	leftFootPosition.y += velocity.y * FOOT_Y_CORRECTION_WEIGHT;
} else {
	leftFootPosition.y -= velocity.y * FOOT_Y_CORRECTION_WEIGHT;
}'

to be changed to this:

`float correction = velocity.y * FOOT_Y_CORRECTION_WEIGHT;
leftFootPosition.y += (velocity.y * leftFootDif.y > 0) ? correction : (correction * -1);'

I just want to be clear about far you want me to take this change.

Reply:
nah, I'm good if you only simplify the easy ones